extras/indexing/src/main/java/org/apache/rya/indexing/mongodb/AbstractMongoIndexer.java

```diff
@@ -68,15 +73,28 @@
 
     protected T storageStrategy;
 
+    private MongoDbBatchWriter<DBObject> mongoDbBatchWriter;
+
     protected void initCore() {
         dbName = conf.get(MongoDBRdfConfiguration.MONGO_DB_NAME);
         db = this.mongoClient.getDB(dbName);
-        collection = db.getCollection(conf.get(MongoDBRdfConfiguration.MONGO_COLLECTION_PREFIX, "rya") + getCollectionName());
+        final String collectionName = conf.get(MongoDBRdfConfiguration.MONGO_COLLECTION_PREFIX, "rya") + getCollectionName();
```

Comment:
Can you modify the AbstactMongoIndexer so that it and the DAO share the same batch writer?